### DISCUSSION TITLE
One or more localized templates contain a set of placeholders (HTML or template literal expressions) that do not exactly match the source code

### DISCUSSION THREAD

**Discussion by pika11** [lit/lit#2712] [Q&A] [answered]
We're getting this error when building translations if a change to an expression was made.

ie. `msg(`Text to translate - ${function.name}` -> `msg(`Text to translate - ${function.name()}`

Adding the () or changing anything inside the expression causes this error. The way to solve this seems to be to change the expression name in all .xlf files but with many of those doesn't seem like the best approach.

Is there a way to either consider that as a new string or change the expression inside in all templates?

**Comment by aomarks** [comment, marked as answer]
Can you paste the entire error message? There should be slightly more after "Details: ".

We recently relaxed a restriction that is related to this (https://github.com/lit/lit/issues/2140), but I guess it doesn't exactly cover this case.

I think that we should relax the restriction further so that this is not an error.

**Reply by pika11** [reply]
Yeah here's what comes next, a bit redacted following the example above:

```
Placeholder error in de-DE localization of s6a...{ID}: unexpected "${function.name}"
Placeholder error in de-DE localization of s6a{ID}: missing "${this.commentCount()}"
```
There would be one of these for each locale that we support

**Reply by aomarks** [reply]
Thanks!

I filed https://github.com/lit/lit/issues/2713 to make this validation less strict.

In the short term, you may be able to directly modify your translated `.xlf` files to make the expressions match up.

**Reply by pika11** [reply]
Yeah that's what we've been doing. Great thank you! I'll be following that issue
